### PR TITLE
feat: updated @aws-sdk/* dependencies 

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:
- Updated @aws-sdk/* dependencies to the most recent version (3.186.0)
- Changed from caret to exact versioning to prevent mismatches in lock file

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/environments/src/postDeployment/cognitoSetup.test.ts [hunk 1]
```diff
@@ -88,7 +89,8 @@ describe('CognitoSetup', () => {
       const ResponseMetadata = { httpStatusCode: 500 };
       cognitoMock.on(AdminCreateUserCommand).rejects(
         new UsernameExistsException({
-          $metadata: ResponseMetadata
+          $metadata: ResponseMetadata,
```
Comment (SanketD92) [inline workbench-core/environments/src/postDeployment/cognitoSetup.test.ts:92]: rename to `responseMetadata`

**Review by SanketD92 — APPROVED** [review]
LGTM

File: workbench-core/environments/src/postDeployment/cognitoSetup.test.ts [hunk 2]
```diff
@@ -88,7 +89,8 @@ describe('CognitoSetup', () => {
       const ResponseMetadata = { httpStatusCode: 500 };
```
Comment (bhudnell) [inline workbench-core/environments/src/postDeployment/cognitoSetup.test.ts:89]:
```suggestion
      const responseMetadata = { httpStatusCode: 500 };
```

File: workbench-core/environments/src/postDeployment/cognitoSetup.test.ts (hunk 1 above)
Comment (bhudnell) [inline workbench-core/environments/src/postDeployment/cognitoSetup.test.ts:92]:
```suggestion
          $metadata: responseMetadata,
```

File: workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.test.ts [hunk 3]
```diff
@@ -711,7 +713,9 @@ describe('CognitoAuthenticationPlugin tests', () => {
     });
 
     it('should throw PluginConfigurationError when the service doesnt have correct permissions', async () => {
-      cognitoMock.on(DescribeUserPoolClientCommand).rejects(new ResourceNotFoundException({ $metadata: {} }));
+      cognitoMock
+        .on(DescribeUserPoolClientCommand)
+        .rejects(new ResourceNotFoundException({ $metadata: {}, message: '' }));
```
Comment (nguyen102) [inline workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.test.ts:718]: Hmm, why did the tests change? Does the new version of AWS sdk add the message field?

File: workbench-core/environments/src/postDeployment/cognitoSetup.test.ts (hunk 1 above)
Comment (bhudnell) [reply workbench-core/environments/src/postDeployment/cognitoSetup.test.ts]: done

File: common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-environments",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 1.186.0",
```
Comment (nguyen102) [inline common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]: this should be 3.186.0 right? Same for the other change files.

File: workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.test.ts (hunk 3 above)
Comment (bhudnell) [reply workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.test.ts:718]: Yeah, it changed message from being optional to being required. I looked through the change logs and didnt see any other behavior changes reported

File: common/changes/@aws/workbench-core-accounts/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-accounts",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 1.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-accounts/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```

File: common/changes/@aws/workbench-core-authentication/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-authentication",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 1.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-authentication/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```

File: common/changes/@aws/workbench-core-base/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-base",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 1.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-base/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```

File: common/changes/@aws/workbench-core-datasets/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-datasets",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 1.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-datasets/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```

File: common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json (hunk 4 above)
Comment (bhudnell) [inline common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```

Comment (bhudnell) [reply common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json]: Oops, yes. Fixed

**Review by nguyen102 — DISMISSED** [review]

**Review by DaveMadsen — CHANGES_REQUESTED** [review]
Can we switch to caret on this? I think caret will be a forcing function to ensure this gets done at the times it needs to happen:
1. Just before we cut a release, we should do a full package update to maximize customer lead times before they need to pull new versions to deal with vulnerabilities/deprecation.
2. When we add a new `@aws-sdk/*` package, we will be out of sync anyway and the update will need to happen.

I would prefer to see us use caret here to ensure this happens for 1 above, and 2 suggests that not using caret doesn't really help us. Note, if we add new sdk packages frequently, this may be an issue, but are we really doing that this often?

File: common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-environments",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```
Comment (nguyen102) [inline common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]: update comment

**Review by DaveMadsen — DISMISSED** [review]

File: common/changes/@aws/workbench-core-accounts/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 10]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-accounts",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-accounts/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies to ^3.186.0",
```

File: common/changes/@aws/workbench-core-authentication/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-authentication",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-authentication/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies to ^3.186.0",
```

File: common/changes/@aws/workbench-core-base/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-base",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-base/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies to ^3.186.0",
```

File: common/changes/@aws/workbench-core-datasets/topic-aws-sdk-package-updates_2022-10-14-00-28.json [hunk 13]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@aws/workbench-core-datasets",
+      "comment": "Updated @aws-sdk/* dependencies and locked version at 3.186.0",
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-datasets/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies to ^3.186.0",
```

File: common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json (hunk 9 above)
Comment (bhudnell) [inline common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json:5]:
```suggestion
      "comment": "Updated @aws-sdk/* dependencies to ^3.186.0",
```

Comment (bhudnell) [reply common/changes/@aws/workbench-core-environments/topic-aws-sdk-package-updates_2022-10-14-00-28.json]: done

**Review by nguyen102 — APPROVED** [review]

**Review by DaveMadsen — APPROVED** [review]

**Pull request merged**